packages/wallets/src/api/client.ts

```diff
@@ -0,0 +1,223 @@
+import { CrossmintApiClient } from "@crossmint/common-sdk-base";
+
+import { SDK_NAME, SDK_VERSION } from "../utils/constants";
+
+import type {
+    CreateWalletDto,
+    WalletV1Alpha2ResponseDto,
+    CreateTransactionDto,
+    SubmitApprovalDto,
+    WalletsV1ControllerGetTransaction4Response,
+    WalletsV1Alpha2TransactionResponseDto,
+    CreateSignatureRequestDto,
+    WalletsV1Alpha2SignatureResponseDto,
+    WalletsV1Alpha2TransactionsResponseDto,
+    Nftevm,
+    Nftsol,
+    WalletBalanceResponseDto,
+} from "./gen/types.gen";
+
+type CreateWalletParams = CreateWalletDto;
+type CreateWalletResponse = WalletV1Alpha2ResponseDto;
+type GetWalletResponse = WalletV1Alpha2ResponseDto;
+
+type CreateTransactionParams = CreateTransactionDto;
+type CreateTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+type ApproveTransactionParams = SubmitApprovalDto;
+type ApproveTransactionResponse = WalletsV1ControllerGetTransaction4Response;
+type GetTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+
+type CreateSignatureParams = CreateSignatureRequestDto;
+type CreateSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type ApproveSignatureParams = SubmitApprovalDto;
+type ApproveSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type GetSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+
+type GetTransactionsResponse = WalletsV1Alpha2TransactionsResponseDto;
+type GetNftsResponse = Nftevm | Nftsol;
+type GetBalanceResponse = WalletBalanceResponseDto;
+
+class ApiClient {
```

Comment:
why not extend `CrossmintApiClient`?

Reply:
that makes total sense https://github.com/Crossmint/crossmint-sdk/pull/956/commits/3059e57e1a362d5dea5e0b57cd123cb2219f5587